src/test/func/redblack/redblack.cc

```diff
@@ -41,37 +67,38 @@ class Rep
   using key = size_t;
 
   static constexpr key null = 0;
+  static constexpr key root = (Wrapper::offset << 1);
 
   using Holder = Wrapper;
   using Contents = size_t;
 
-  static void set(Holder* ptr, Contents r)
+  static void set(Holder ptr, Contents r)
   {
-    ptr->value = ((r + Wrapper::offset) << 1) + (ptr->value & 1);
+    ptr.set(r);
   }
 
-  static Contents get(Holder* ptr)
+  static Contents get(Holder ptr)
   {
-    return (ptr->value >> 1) - Wrapper::offset;
+    return (Contents)ptr;
```

Comment:
Best not to use C-style casts.  Though I don't think Clang-tidy checks the tests, so they don't all meet this bar.

Reply:
We currently run clang-tidy only on the includes.  We probably should extend it to the tests.